services/provider/server/server.go

```diff
@@ -741,17 +742,43 @@ func (s *OCSProviderServer) isSubscriptionChannelValid(ctx context.Context, subs
 	}
 
 	// Check if the installed CSV is the expected one for the channel from the package manifest
-	packageManifest := &unstructured.Unstructured{}
-	packageManifest.SetGroupVersionKind(schema.GroupVersionKind{
+	packageManifestList := &unstructured.UnstructuredList{}
+	packageManifestList.SetGroupVersionKind(schema.GroupVersionKind{
 		Group:   "packages.operators.coreos.com",
 		Version: "v1",
 		Kind:    "PackageManifest",
 	})
-	err = s.client.Get(ctx, client.ObjectKey{Name: subscription.Spec.Package, Namespace: s.namespace}, packageManifest)
+	err = s.client.List(ctx, packageManifestList, client.InNamespace(s.namespace))
 	if err != nil {
-		klog.Errorf("failed to get PackageManifest %q: %v", subscription.Spec.Package, err)
+		klog.Errorf("failed to list packageManifests for package %q: %v", packageName, err)
 		return false, err
 	}
+	var packageManifest *unstructured.Unstructured
+	for i := range packageManifestList.Items {
+		pm := &packageManifestList.Items[i]
+		if pm.GetName() != packageName {
+			continue
+		}
+		labels := pm.GetLabels()
+		if labels == nil {
+			continue
+		}
+		catalogLabel, catalogExists := labels["catalog"]
+		if !catalogExists || catalogLabel != subscription.Spec.CatalogSource {
+			continue
+		}
+		catalogNamespaceLabel, catalogNamespaceExists := labels["catalog-namespace"]
+		if !catalogNamespaceExists || catalogNamespaceLabel != subscription.Spec.CatalogSourceNamespace {
+			continue
+		}
+		packageManifest = pm
+		break
+	}
```

Comment:
Suggestion only:
Given all of the other comments above
This entire body can be written more compactly without the need for a `continue` jump
 ```suggestion
	for i := range packageManifestList.Items {
		pm := &packageManifestList.Items[i]
		labels := pm.GetLabels()
		
		if pm.GetName() != packageName &&
		   	labels["catalog"] != subscription.Spec.CatalogSource && 
		    	labels["catalog-namespace"] != subscription.Spec.CatalogSourceNamespace {
		    	
		    	packageManifest = pm
			break
	        }
	}
```